### PR TITLE
create-turbo: automatic git configuration.

### PR DESCRIPTION
**Pull request by nathanhammond**
Not all environments where `create-turbo` is run will have a preconfigured `git` config. This ensures that we always have an author config for the initial commit.

It also reduces the number of processes executed to reduce the chances of intermittent failures.

### CODE REVIEW

File: packages/create-turbo/src/utils/git.ts [hunk 1]
```diff
@@ -51,7 +51,6 @@ export function isInMercurialRepository(): boolean {
 export function tryGitInit(root: string, message: string): boolean {
   let didInit = false;
   try {
-    execSync("git --version", { stdio: "ignore" });
```
Comment (nathanhammond) [inline packages/create-turbo/src/utils/git.ts:54]: This is unnecessary; just a bonus place where weird errors can creep in.

File: packages/create-turbo/src/utils/git.ts [hunk 2]
```diff
@@ -78,12 +74,18 @@ export function tryGitInit(root: string, message: string): boolean {
...
 }
+
+function gitCommit(message: string) {
+  execSync(
+    `git commit --author="Turbobot <turbobot@vercel.com>" -am "${message}"`,
+    {
+      stdio: "ignore",
+    }
+  );
+}
```
Comment (nathanhammond) [inline packages/create-turbo/src/utils/git.ts:91]: Factored out into a separate method so you get either the `try` behavior, or the `boolean` behavior as you see fit.

File: packages/create-turbo/src/utils/git.ts [hunk 3]
```diff
@@ -78,12 +74,18 @@ export function tryGitInit(root: string, message: string): boolean {
...
+    gitCommit(message);
     return true;
   } catch (err) {
     return false;
   }
 }
+
+function gitCommit(message: string) {
+  execSync(
+    `git commit --author="Turbobot <turbobot@vercel.com>" -am "${message}"`,
```
Comment (nathanhammond) [inline packages/create-turbo/src/utils/git.ts:86]: Hardcodes the automated commits to be from us: `Turbobot <turbobot@vercel.com>`

**Review by tknickman — APPROVED** [review]
Needs test updates, but lgtm

**Pull request merged**